### PR TITLE
[FEAT] Custom content renders for month, quarter and year

### PR DESCRIPTION
**Pull request by KevinFabre-ods**
I work for an organization that requires this UI for day, month and year elements in selected / selecting range.

| Day | Month | Year |
|---|---|---|
|Screenshot 2023-06-22 at 15 35 47|Screenshot 2023-06-22 at 15 42 29|Screenshot 2023-06-23 at 11 19 34|

To do this UI, it required to use custom content renders:
- One div `.react-datepicker__{precision}` for the background.
- Another div `.react-datepicker__{precision}-content` to wrap the text in some kind of chip.

This PR aims to offer custom render for month, quarter and year content in the same way as custom day content render.

### Question:

- Should these props be singular or plural ? renderYearContent / renderYearContent**s**
_I prefer the singular form as the element has only one content._

### CODE REVIEW

**Comment by martijnrusschen** [comment]
> Should these props be singular or plural ? renderYearContent / renderYearContents
> I prefer the singular form as the element has only one content.

Agreed, I think that makes more sense.

**Comment by KevinFabre-ods** [comment]
> 



> > Should these props be singular or plural ? renderYearContent / renderYearContents
> > I prefer the singular form as the element has only one content.
> 
> Agreed, I think that makes more sense.

So, maybe it's a good time to rename renderDayContents to renderDayContent ?

File: src/date_utils.js [hunk 1]
```diff
@@ -8,8 +8,6 @@ import addWeeks from "date-fns/addWeeks";
 import addMonths from "date-fns/addMonths";
 import addQuarters from "date-fns/addQuarters";
 import addYears from "date-fns/addYears";
-import subMinutes from "date-fns/subMinutes";
```
Comment (martijnrusschen) [inline src/date_utils.js:11]: I assume this is just clean up?

File: src/index.jsx [hunk 2]
```diff
@@ -119,10 +119,6 @@ export default class DatePicker extends React.Component {
       timeInputLabel: "Time",
       enableTabLoop: true,
       yearItemNumber: DEFAULT_YEAR_ITEM_NUMBER,
-
-      renderDayContents(date) {
```
Comment (martijnrusschen) [inline src/index.jsx:123]: was this unused?

**Review by martijnrusschen — APPROVED** [review]

**Comment by martijnrusschen** [comment]
> > 
> 
> > > Should these props be singular or plural ? renderYearContent / renderYearContents
> > > I prefer the singular form as the element has only one content.
> > 
> > 
> > Agreed, I think that makes more sense.
> 
> So, maybe it's a good time to rename renderDayContents to renderDayContent ?

sounds good to me.

**Comment by martijnrusschen** [comment]
> > > 
> > 
> > 
> > > > Should these props be singular or plural ? renderYearContent / renderYearContents
> > > > I prefer the singular form as the element has only one content.
> > > 
> > > 
> > > Agreed, I think that makes more sense.
> > 
> > 
> > So, maybe it's a good time to rename renderDayContents to renderDayContent ?
> 
> sounds good to me.

Only concern is that it may be  a breaking change. Perhaps leave that for now, if it's breaking the API.

File: src/index.jsx (hunk 2 above)
Comment (KevinFabre-ods) [reply src/index.jsx:123]: Yes, as there is [already a check](https://github.com/Hacker0x01/react-datepicker/blob/e67fe3a6f46d27e8765b9193f1ec90a383dcad0e/src/day.jsx#L343) if this prop is defined.

```javascript
   return this.props.renderDayContents
      ? this.props.renderDayContents(getDate(this.props.day), this.props.day)
      : getDate(this.props.day);
```

File: src/date_utils.js (hunk 1 above)
Comment (KevinFabre-ods) [reply src/date_utils.js:11]: Yes, during build there were three warnings about unused imports

Comment (martijnrusschen) [reply src/date_utils.js:11]: ah great, should probably implement a linter for that :)

**Pull request merged**